From 4a9e1fac5d69e6abae0451c579b02f16d960e694 Mon Sep 17 00:00:00 2001
Message-Id: <4a9e1fac5d69e6abae0451c579b02f16d960e694.1471442559.git.mleitner@redhat.com>
From: Xin Long <lucien.xin@gmail.com>
Date: Thu, 11 Aug 2016 15:11:38 +0800
Subject: [PATCH] teamd: fix the issue that network blocks when systemctl stop
 teamd

Commit 0641375d10d6 ("teamd: change to Before=network-pre.target in
systemd service file") tried to make teamd be stopped after network
service is stopped when systemd shutdown.

But network service also kills teamd service with "systemctl stop" in
ifdown-Team. It means network service try to kill one service that is
dependent on network itself, which leads to a block.

This patch is to use "systemctl stop --ignore-dependencies" to avoid
this block without any side effects. Already verified it on the env
where we can reproduce this issue.

Fixes: 0641375d10d6 ("teamd: change to Before=network-pre.target in systemd service file")
Signed-off-by: Xin Long <lucien.xin@gmail.com>
Signed-off-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
---
 teamd/redhat/initscripts_systemd/network-scripts/ifdown-Team | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/teamd/redhat/initscripts_systemd/network-scripts/ifdown-Team b/teamd/redhat/initscripts_systemd/network-scripts/ifdown-Team
index 9404e05ffa89051332bed22850d9fd0722efc644..73eccee188773f3493077fe80fbaddacd8ee0226 100755
--- a/teamd/redhat/initscripts_systemd/network-scripts/ifdown-Team
+++ b/teamd/redhat/initscripts_systemd/network-scripts/ifdown-Team
@@ -45,5 +45,5 @@ if [ -n "${TEAM_CONFIG}" ]; then
 		is_ignored_file "$device" && continue
 		/sbin/ifdown ${device##*/}
 	done
-	/usr/bin/systemctl stop teamd@${DEVICE}.service || exit 1
+	/usr/bin/systemctl stop teamd@${DEVICE}.service --ignore-dependencies || exit 1
 fi
-- 
2.7.4

